tests/test_tex.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+# Copyright 2012-2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import print_function
+
+import operator
+import unittest
+
+import numpy as np
+
+import lsst.utils
+from lsst.validate.drp.calcsrd.tex import select_bin_from_corr
+
+
+class TexCalculations(lsst.utils.tests.TestCase):
+    """Test calculation of TEx ellipticity residuals calculations."""
+
+    def testSelectBinFromCorr(self):
+        """Does select_bin_from_corr correctly return only and all the bins that satisfy condition."""
+        r = np.array([0.1, 0.2, 0.5, 0.8, 1.0, 1.2, 1.5, 2.0, 2.5])
+        xip = 1e-5 * np.array([0.9, 1.1, 1.5, 0.02, 1.2, 10.0, 5.0, 2.5, 3.0])
+        xip_err = 1e-6 * np.array([1, 1, 1, 1, 1, 2, 2, 2, 2])
+
+        exp_all_avg_xip, exp_all_avg_xip_err = np.average(xip), np.average(xip_err)
+        obs_all_avg_xip, obs_all_avg_xip_err = \
+            select_bin_from_corr(r, xip, xip_err, radius=0, operator=operator.gt)
+        self.assertFloatsAlmostEqual(exp_all_avg_xip, obs_all_avg_xip, rtol=1e-7)
+        self.assertFloatsAlmostEqual(exp_all_avg_xip_err, obs_all_avg_xip_err, rtol=1e-7)
+
+        exp_rlt2_avg_xip, exp_rlt2_avg_xip_err = 2.8171429e-05, 1.2857143e-06
+        obs_rlt2_avg_xip, obs_rlt2_avg_xip_err = \
+            select_bin_from_corr(r, xip, xip_err, radius=2, operator=operator.lt)
+        self.assertFloatsAlmostEqual(exp_rlt2_avg_xip, obs_rlt2_avg_xip, rtol=1e-7)
+        self.assertFloatsAlmostEqual(exp_rlt2_avg_xip_err, obs_rlt2_avg_xip_err, rtol=1e-7)
+
+        exp_rge1_avg_xip, exp_rge1_avg_xip_err = 4.3400000e-05, 1.8000000e-06
+        obs_rge1_avg_xip, obs_rge1_avg_xip_err = \
+            select_bin_from_corr(r, xip, xip_err, radius=1, operator=operator.ge)
+        self.assertFloatsAlmostEqual(exp_rge1_avg_xip, obs_rge1_avg_xip, rtol=1e-7)
+        self.assertFloatsAlmostEqual(exp_rge1_avg_xip_err, obs_rge1_avg_xip_err, rtol=1e-7)
+
```

Comment:
I think you also want some tests for your `correlation_function_ellipticity*` functions, and probably for the added `utils` functions.